docs: Improve documentation for Result (#14)
taxfyle/FxKit: docs/core/result.md

}
```

### Unwrap

> The `Unwrap` / `UnwrapOrThrow` gets the `Ok` value, or throws an exception otherwise.

Reviewer: ```suggestion
The `Unwrap` / `UnwrapOrThrow` methods get the `Ok` value, or throw an exception otherwise.
```

Author: Done.

```csharp
Result<int, string> result = Divide(10, 2);
int resultValue = result.Unwrap("Optional error message");
```

> Or you can use `UnwrapOr` / `UnwrapOrElse` and specify a fallback value for when result is in error state.

Reviewer: ```suggestion
Or you can use `UnwrapOr` / `UnwrapOrElse` and specify a fallback value for when the result is in the error state.
```

Author: Done.

```csharp
Result<int, string> result = Divide(10, 2);
int resultValue = result.UnwrapOr(0);
```

### UnwrapErr

> The `UnwrapErr` gets the `Err` value, or throws an exception otherwise.

Reviewer: ```suggestion
The `UnwrapErr` method gets the `Err` value, or throws an exception otherwise.
```

Author: Done.

```csharp
Result<int, string> result = Divide(10, 0);
string error = result.UnwrapErr("Optional error message");
```

### UnwrapEither

> The `UnwrapEither` returns either the value or the error when `Result` value type is the same of error type.

Reviewer: ```suggestion
The `UnwrapEither` method returns either the Ok value or the Err value when both are of the same type.
```

Author: Done.
